File: src/main/java/problems/basic/LCMandGCD/LCM_and_GCD.py
# User function Template for python3

class Solution:
    def lcmAndGcd(self, A, B):
        # code here
        return [self.lowest_common_multiple(A, B), self.greatest_common_divisor(A, B)]

    def greatest_common_divisor(self, A: int, B: int) -> int:
        while B != 0:
            remainder = A % B
            if remainder == 0:
                return B
            A = B
            B = remainder
        return A

    def lowest_common_multiple(self, A: int, B: int) -> int:
        gcm = self.greatest_common_divisor(A, B)
        return int(A / gcm * B)


'''
#{ 
#  Driver Code Starts
#Initial Template for Python 3

import math
if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        A,B=map(int,input().split())
        
        ob = Solution()
        ptr = ob.lcmAndGcd(A,B)
        
        print(ptr[0],ptr[1])
# } Driver Code Ends
'''

File: src/main/java/problems/basic/LCMandGCD/LCM_and_GCD.java
package problems.basic.LCMandGCD;
// { Driver Code Starts

import java.io.*;

@SuppressWarnings("PMD")
class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String S[] = read.readLine().split(" ");
            Long A = Long.parseLong(S[0]);
            Long B = Long.parseLong(S[1]);

            Solution ob = new Solution();
            Long[] ptr = ob.lcmAndGcd(A, B);

            System.out.print(ptr[0]);
            System.out.print(" ");
            System.out.println(ptr[1]);
        }
    }
} // } Driver Code Ends

@SuppressWarnings("PMD")
class Solution {
    static Long[] lcmAndGcd(Long A, Long B) {
        // code here
        return new Long[] {};
    }
}
